### PR TITLE
Consolidate SetupHandlers

### LINKED ISSUE #5150: Review Console and Backtesting ISetupHandler
#### Expected Behavior

- ISetupHandler implementations are not duplicated
#### Actual Behavior

- Console and Backtesting setup handlers are both used for backtesting, there is a lot of code duplication among them, what is the reason we need both, are they different?
#### Potential Solution

- Review and consolidate Console & backtesting setup handler
#### Reproducing the Problem

N/A
#### System Information

N/A
#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by C-SELLERS**
#### Description

- Reduce duplicated code between setuphandlers by consolidating BacktestingSetupHandler and ConsoleSetupHandler

#### Related Issue

Closes #5150 

#### Motivation and Context

These two handlers were almost identical in implementation. The only important unique factors about them are their MaxOrder and MaximumRuntime variables which are set differently. The address this difference I used virtual functions `GetMaximumOrders` and `GetMaximumRuntime`.

#### Requires Documentation Change

N/A

#### How Has This Been Tested?

WIP

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by C-SELLERS** [comment]
One thing that was **NOT** in `BacktestingSetupHandler` that was in `ConsoleSetupHandler` was a call to set `algorithm.SetAvailableDataTypes(GetConfiguredDataFeeds());`. I went ahead and added this to `BacktestingSetupHandler.Setup()` because it just updates the algorithms acceptable ticktypes for each security type, and if none are in the config it does nothing. Can confirm that it loads in default values anyways.

**Comment by C-SELLERS** [comment]
It appears after adding `SetAvailableDataTypes` we are failing to pass all regressions, will address asap.

File: Engine/Setup/BaseSetupHandler.cs [hunk 1]
```diff
@@ -152,5 +153,23 @@ public static void LoadBacktestJobAccountCurrency(IAlgorithm algorithm, Backtest
...
+        /// Get the available data feeds from config.json,
+        /// If none available, throw an error
+        /// </summary>
+        public static Dictionary<SecurityType, List<TickType>> GetConfiguredDataFeeds()
+        {
+            var dataFeedsConfigString = Config.Get("security-data-feeds");
+
+            if (!dataFeedsConfigString.IsNullOrEmpty())
+            {
+                Dictionary<SecurityType, List<TickType>> dataFeeds = new Dictionary<SecurityType, List<TickType>>();
```
Comment (Martin-Molinero) [inline Engine/Setup/BaseSetupHandler.cs:167]: nit - don't think this line is required? The `new` resulting object will be overriden bellow

File: Engine/Setup/BacktestingSetupHandler.cs [hunk 2]
```diff
@@ -262,17 +245,51 @@ public bool Setup(SetupHandlerParameters parameters)
...
+            {
+                MaxOrders = 10000;
+            }
+            else
+            {
+                MaxOrders = int.MaxValue;
+                MaximumRuntime += MaximumRuntime;
+            }
+
+            MaxOrders = job.Controls.BacktestingMaxOrders;
```
Comment (Martin-Molinero) [inline Engine/Setup/BacktestingSetupHandler.cs:271]: This looks fishy, `MaxOrders` value is overriden by `job.Controls.BacktestingMaxOrders` so what's the `if/else` above for? `MaximumRuntime += MaximumRuntime;` is also fishy 😁 should be in the call to `GetMaximumRuntime()` I think

File: Engine/Setup/BaseSetupHandler.cs (hunk 1 above)
Comment (C-SELLERS) [reply Engine/Setup/BaseSetupHandler.cs:167]: Missed that in the refactor when I let it pass null back, ty 😁

File: Engine/Setup/BacktestingSetupHandler.cs (hunk 2 above)
Comment (C-SELLERS) [reply Engine/Setup/BacktestingSetupHandler.cs]: 💯

**Review by Martin-Molinero — APPROVED** [review]
Awesome consolidation!

**Pull request merged**